kong/dao/migrations/helpers.lua

```diff
@@ -95,5 +99,205 @@ function _M.plugin_config_iterator(dao, plugin_name)
+destination_table_def. Both table_defs have the following structure
+  { name    = "ssl_certificates",
+    columns = {
+      id         = "uuid",
+      cert       = "text",
+      key        = "text",
+      created_at = "timestamp",
+    },
+    partition_keys = { "id" },
+  }
+
+columns_to_copy is a hash-like table.
+* Each key must be a string D representing a column in the destination table.
+* If the value is a string S, then the value of the S column in the source
+  table will be assigned to the D column in destination.
+* If the value is a function, then the result of executing it will be assigned
+  to the D column.
+Example:
+  {
+    partition  = function() return cassandra.text("certificates") end,
+    id         = "id",
+    cert       = "cert",
+    key        = "key",
+   created_at = "created_at",
+  }
+
+The function takes the "source row" as parameter, so it could be used to do things
+like merging two fields together into one, or putting a string in uppercase.
+--]]
+function _M.cassandra.copy_records(dao,
+                                   source_table_def,
+                                   destination_table_def,
+                                   columns_to_copy)
+
+  local coordinator = dao.db:get_coordinator()
+  local cql = fmt("SELECT * FROM %s ALLOW FILTERING", source_table_def.name)
+  for rows, err in coordinator:iterate(cql) do
+    if err then
+      return nil, err
+    end
+
+    for _, source_row in ipairs(rows) do
+      local column_names = {}
+      local values = {}
+      local len = 0
+
+      for dest_column_name, source_value in pairs(columns_to_copy) do
+        if type(source_value) == "string" then
+          source_value = source_row[source_value]
+          local dest_type = destination_table_def.columns[dest_column_name]
+          local type_converter = cassandra[dest_type]
+          if not type_converter then
+            return nil, fmt("Could not find the cassandra type converter for column %s (type %s)",
+                            dest_column_name, source_table_def[dest_column_name])
+          end
+          source_value = type_converter(source_value)
+
+        elseif type(source_value) == "function" then
+          source_value = source_value(source_row)
+        else
+          return nil, fmt("Expected a string or function, found %s (a %s)",
+                          tostring(source_value), type(source_value))
+        end
+
+        if source_value ~= nil then
+          len = len + 1
+          values[len] = source_value
+          column_names[len] = dest_column_name
+        end
+      end
+
+      local question_marks = string.sub(string.rep("?, ", len), 1, -3)
+
+      local insert_cql = fmt("INSERT INTO %s (%s) VALUES (%s)",
+                             destination_table_def.name,
+                             table_concat(column_names, ", "),
+                             question_marks)
+      local _, err = coordinator:execute(insert_cql, values, nil, "write")
```

Comment:
`execute()` from lua-cassandra only takes `(query, args, options)` as arguments, need to remove the two trailing ones, or eventually pass `{ consistency = cql.consistencies.all }` (as do other migrations).